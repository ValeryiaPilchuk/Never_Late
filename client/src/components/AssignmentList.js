import * as React from 'react';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';





function createData(Assignment_name, start_time, due_time) {
  return { Assignment_name, start_time, due_time };
}

// Dummy 
const rows = [
  createData('Assignment 1', "Nov 9 10:00 am", "Nov 10 10:00 am"),
  createData('Assignment 2', "Nov 9 10:00 am", "Nov 13 10:00 am"),
  createData('Assignment 3', "Nov 9 10:00 am", "Nov 14 10:00 am"),
  createData('Assignment 4', "Nov 9 10:00 am", "Nov 15 10:00 am"),
  createData('Assignment 5', "Nov 9 10:00 am", "Nov 16 10:00 am"),
 

];

export default function AssignmentList() {
  return (
    <TableContainer component={Paper}>

      <Table  >

        <TableHead sx = {{bgcolor:'#fffbf2'}}>
          <TableRow  >
            <TableCell  sx ={{fontWeight:'600', color:'primary.light' }} >Assigment Name</TableCell>
            <TableCell sx ={{fontWeight:'600', color:'primary.light' }} align="right">Start Time</TableCell>
            <TableCell sx ={{fontWeight:'600', color:'primary.light'}} align="right">Due Time</TableCell>
          </TableRow>
        </TableHead>


        <TableBody>
          {rows.map((row) => (
            <TableRow
              key={row.Assignment_name}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row"> {row.Assignment_name}</TableCell>
              <TableCell align="right">{row.start_time}</TableCell>
              <TableCell align="right">{row.due_time}</TableCell>
            </TableRow>
          ))}
        </TableBody>

      </Table>

    </TableContainer>
  );
}
