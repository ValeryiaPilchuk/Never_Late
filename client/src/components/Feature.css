.feature-title{
    color: black;
    font-size: 1.5rem;
}

p{
    color: gray;
}