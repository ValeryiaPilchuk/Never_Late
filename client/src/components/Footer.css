.footer {
  background-color: rgb(202, 205, 248);
  padding: 20px;
}
