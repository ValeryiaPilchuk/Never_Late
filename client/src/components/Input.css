.drop-zone {
  max-width: 300px;
  height: 300px;
  padding: 25px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  font-weight: 500;
  font-size: 30px;
  cursor: pointer;
  color: rgb(202, 205, 248);
  border: 4px dashed #229954 ;
  border-radius: 10px;
  margin: auto;
}

.drop-zone--over {
  border-style: solid;
}

.drop-zone__input {
  display: none;
}

.drop-zone__thumb {
  width: 100%;
  height: 100%;
  border-radius: 10px;
  overflow: hidden;
  background-color: rgb(202, 205, 248);
  background-size: cover;
  position: relative;
}

.drop-zone__thumb::after {
  content: attr(data-label);
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  padding: 5px 0;
  background: rgba(0, 0, 0, 0.75);
  font-size: 14px;
  text-align: center;
}
.message {
  max-width: 500px;
  padding: 25px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  font-weight: 500;
  font-size: 20px;
  color: rgb(202, 205, 248);
  border-radius: 10px;
  margin: auto;
}

.upload_button {
  max-width: 400px;
  height: 50px;
  padding: 25px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  font-weight: 500;
  font-size: 20px;
  color: rgb(0, 100, 0);
  border-radius: 10px;
  margin: auto;
  background-color: rgb(220,220,220);
;
}