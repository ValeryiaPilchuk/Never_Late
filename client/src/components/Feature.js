import React from 'react'
import "./Feature.css";
import { Grid, Paper } from '@mui/material';
import logo from '../static/NeverLate-logos.jpeg';



export default function Feature(){

    
    return(
    <>
   
    <img src={logo} alt="" className="logo1" />


    <Grid  sx={{padding:"7% 8%"}}  container spacing={8}  >
        <Grid item  md={4} >
            <Paper sx= {{textAlign: "center", padding:"5%", background:'#D8AED3'}}> 
                
                <h3 className ="feature-title">Easy to Start </h3>
                <p>Just connect with your BlackBoard account and the website will automatically update the assignment due day   </p>
             </Paper>
        </Grid>     
        <Grid item  md={4}>
            <Paper sx= {{textAlign: "center", padding:"5%", background:'#D8AED3'}}> 
                
                <h3 className ="feature-title">Good Reminder</h3>
                <p>The website will sent out email or phone message to remind you the due day of the assignment</p>
             </Paper>
        </Grid>      
        <Grid item  md={4}>
            <Paper sx= {{textAlign: "center", padding:"5%", background:'#D8AED3'}}> 
                
                <h3 className ="feature-title">User Friendly</h3>
                <p> The website interface is easy for the user to use </p>
             </Paper>
        </Grid>         
    </Grid>

  
    </>
    )
}