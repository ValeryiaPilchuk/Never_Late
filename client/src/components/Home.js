import React from 'react'
import './HomePage.css';
import Feature from './Feature';


export default function Home() {
    return (
        <>
        <div   data-testid='h-1' className="homePage">
            <br/><br/><br/>
            
            <div data-testid='h-2' className= "title">
            <h1>
                Welcome to NeverLate 
            </h1>
                <h3>The best choice for your homework schedule mangement. </h3>
            </div>  
            
            </div>
            <Feature data-testid='h-3' />
        </>
    )
}
