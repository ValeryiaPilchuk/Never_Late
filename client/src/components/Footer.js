import React from 'react'
import envelope from '../static/envelope.e7283f94.svg'
import './Footer.css'; 

export default function Footer() {
    return (
        <footer className="footer">
            <div className="content has-text-centered">
                <div className="media">
                    <div className="p-1">
                    </div>
                    <div className="media-body">
                        <div className="h5 p-1">
                            <img src={envelope} alt="contact us" className="contact" />  <strong>Email:</strong> <a href="mailto:email@domain.com">email@domain.com</a>
                        </div>
                    </div>
                </div>
            </div>
        </footer>
    )
}
