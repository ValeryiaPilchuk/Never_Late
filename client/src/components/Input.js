import React from 'react'
import './Input.css';


// Updates the thumbnail on a drop zone element.
function updateThumbnail(dropZoneElement, file) {
  let thumbnailElement = dropZoneElement.querySelector(".drop-zone__thumb");

  // First time - remove the prompt
  if (dropZoneElement.querySelector(".drop-zone__prompt")) {
    dropZoneElement.querySelector(".drop-zone__prompt").remove();
  }

  // First time - there is no thumbnail element, so lets create it
  if (!thumbnailElement) {
    thumbnailElement = document.createElement("div");
    thumbnailElement.classList.add("drop-zone__thumb");
    dropZoneElement.appendChild(thumbnailElement);
  }

  thumbnailElement.dataset.label = file.name;
   // Show thumbnail for image files
  if (file.type.startsWith("image/")) {
    const reader = new FileReader();

    reader.readAsDataURL(file);
    reader.onload = () => {
      thumbnailElement.style.backgroundImage = `url('${reader.result}')`;
    };
  } else {
    thumbnailElement.style.backgroundImage = null;
  }
}


export default function Input() {

function savetoDB() {
    window.location.reload()

    // IndexedDB
    var indexedDB = window.indexedDB || window.webkitIndexedDB || window.mozIndexedDB || window.OIndexedDB || window.msIndexedDB,
        IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.OIDBTransaction || window.msIDBTransaction,
        dbVersion = 1.0;

    // Create/open database
    var request = indexedDB.open("pdfuploads", dbVersion),
        db,
        createObjectStore = function (dataBase) {
            // Create an objectStore
            console.log("Creating objectStore")
            dataBase.createObjectStore("pdf");
        },

        getPdfFile = function () {
            // Create XHR
            var xhr = new XMLHttpRequest(),
                blob;

            xhr.open("GET", "elephant.png", true);
            // Set the responseType to blob
            xhr.responseType = "blob";

            xhr.addEventListener("load", function () {
                if (xhr.status === 200) {
                    console.log("File retrieved");

                    // Blob as response
                    blob = xhr.response;
                    console.log("Blob:" + blob);

                    // Put the received blob into IndexedDB
                    putElephantInDb(blob);
                }
            }, false);
            // Send XHR
            xhr.send();
        },

        putElephantInDb = function (blob) {
            console.log("Putting pdfs in IndexedDB");

            // Open a transaction to the database
            var transaction = db.transaction(["pdf"], IDBTransaction.READ_WRITE);

            // Put the blob into the dabase
            var put = transaction.objectStore("pdf").put(blob, "pdf");

            // Retrieve the file that was just stored
            transaction.objectStore("pdf").get("pdf").onsuccess = function (event) {
                var File = event.target.result;
                console.log("Got pdf!" + File);

                // Get window.URL object
                var URL = window.URL || window.webkitURL;

                // Create and revoke ObjectURL
                var FileURL = URL.createObjectURL(File);

                // Set img src to ObjectURL
                var imgElephant = document.getElementById("pdf");
                imgElephant.setAttribute("src", FileURL);

                // Revoking ObjectURL
                URL.revokeObjectURL(FileURL);
            };
        };

    request.onerror = function (event) {
        console.log("Error creating/accessing IndexedDB database");
    };

    request.onsuccess = function (event) {
        console.log("Success creating/accessing IndexedDB database");
        db = request.result;

        db.onerror = function (event) {
            console.log("Error creating/accessing IndexedDB database");
        };

        // Interim solution for Google Chrome to create an objectStore. Will be deprecated
        if (db.setVersion) {
            if (db.version !== dbVersion) {
                var setVersion = db.setVersion(dbVersion);
                setVersion.onsuccess = function () {
                    createObjectStore(db);
                    getPdfFile();
                };
            }
            else {
                getPdfFile();
            }
        }
        else {
            getPdfFile();
        }
    }
    }
    window.onload = function () {
        document.querySelectorAll(".drop-zone__input").forEach((inputElement) => {
          const dropZoneElement = inputElement.closest(".drop-zone");

          dropZoneElement.addEventListener("click", (e) => {
            inputElement.click();
            console.log("user clicked on input")
          });

          inputElement.addEventListener("change", (e) => {
            if (inputElement.files.length) {
              updateThumbnail(dropZoneElement, inputElement.files[0]);
            }
          });

          dropZoneElement.addEventListener("dragover", (e) => {
            e.preventDefault();
            dropZoneElement.classList.add("drop-zone--over");
          });

          ["dragleave", "dragend"].forEach((type) => {
            dropZoneElement.addEventListener(type, (e) => {
              dropZoneElement.classList.remove("drop-zone--over");
            });
          });

          dropZoneElement.addEventListener("drop", (e) => {
            e.preventDefault();

            if (e.dataTransfer.files.length) {
              inputElement.files = e.dataTransfer.files;
              updateThumbnail(dropZoneElement, e.dataTransfer.files[0]);
            }

            dropZoneElement.classList.remove("drop-zone--over");
          });
        });
    };


  return(
  <>
    <head>
      <title>Drag and Drop File Upload</title>
        <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
        <meta charset="utf-8"/>
      <link rel="shortcut icon" href="/assets/favicon.ico"/>
    </head>
    <body>
      <div class="drop-zone">
        <span class="drop-zone__prompt">Drop file here or click to upload</span>
        <input type="file" name="myFile" class="drop-zone__input"/>
      </div>
      <br></br>
      <button id="btnFileUpload" onclick = "savetoDB()" value="Upload PDF" class = "upload_button">Submit</button>
      <span class="message">Please refresh the page if any issue occurs</span>
    </body>
    </>
  )
}
