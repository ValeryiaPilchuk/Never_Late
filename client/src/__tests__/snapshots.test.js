import Home from "../components/Home";
import {render,cleanup, fireEvent, screen} from '@testing-library/react'
import ReactDom from "react-dom";
import { Dashboard, Login } from "@mui/icons-material";
import {Feature} from "../components/Feature";
import { Footer } from "react-bootstrap/lib/Modal";
import { Navbar } from "react-bootstrap";
import { Signup } from "../components/Signup";
import '@testing-library/jest-dom'
import Profile from "../components/Profile"
import Calendar from "../components/Calendar";
import renderer from "react-test-renderer";


afterEach(cleanup);

test("Profile matches snapshot", ()=>{
    const tree = renderer.create(<Profile></Profile>).toJSON();
    expect(tree).toMatchSnapshot();
})

