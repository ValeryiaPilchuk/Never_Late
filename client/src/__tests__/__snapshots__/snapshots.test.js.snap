// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches snapshot 1`] = `
<div>
  <div
    className="container bootstrap snippets bootdey"
  >
    <div
      className="row ng-scope"
    >
      <div
        className="col-md-4"
      >
        <div
          className="panel panel-default"
        >
          <div
            className="panel-body text-center"
          >
            <div
              className="pv-lg"
            >
              <img
                alt="Contact"
                className="center-block img-responsive img-circle img-thumbnail thumb96"
                src="https://i.redd.it/1krwuhtbr7a71.jpg"
              />
            </div>
            <h3
              className="m0 text-bold"
            >
              Patrick Star
            </h3>
            <div
              className="mv-lg"
            >
              <p>
                Freshman
              </p>
              <p>
                3rd year student
              </p>
              <p>
                Undecided
              </p>
              <p>
                Ready to learn
              </p>
            </div>
          </div>
        </div>
        <div
          className="panel panel-default hidden-xs hidden-sm"
        >
          <div
            className="panel-heading"
          >
            <div
              className="panel-title text-center"
            >
              Enrolled In
            </div>
          </div>
          <div
            className="panel-body"
          >
            <div
              className="media"
            >
              <div
                className="media-left media-middle"
              >
                <img
                  alt="image1"
                  className="media-object img-circle img-thumbnail thumb48"
                  src="https://external-preview.redd.it/IqMKJvPNfvnuKMqE16CvCM--btJpr6pcZ32wtXvnssY.jpg?auto=webp&s=29fc97533ffd3e4adb5e889cc545700810bf4996"
                />
              </div>
              <div
                className="media-body pt-sm"
              >
                <div
                  className="text-bold"
                >
                  Clarinet Practice
                  <div
                    className="text-sm text-muted"
                  >
                    Professor Squidward
                  </div>
                </div>
              </div>
            </div>
            <div
              className="media"
            >
              <div
                className="media-left media-middle"
              >
                <img
                  alt="image2"
                  className="media-object img-circle img-thumbnail thumb48"
                  src="https://images.gamebanana.com/img/ico/sprays/5f804891da197.png"
                />
              </div>
              <div
                className="media-body pt-sm"
              >
                <div
                  className="text-bold"
                >
                  Petrology
                  <div
                    className="text-sm text-muted"
                  >
                    Dr. Rock
                  </div>
                </div>
              </div>
            </div>
            <div
              className="media"
            >
              <div
                className="media-left media-middle"
              >
                <img
                  alt="image3"
                  className="media-object img-circle img-thumbnail thumb48"
                  src="https://lh3.googleusercontent.com/proxy/aT9L0UqWG_c_Aiogpc8USYwKTcaZ7hUL4b7wVMXWryiqWyl1wb8UKZYL5MEJOzc9w30RxO2CNWwFmhqUyZgx7735KqnT0fdHWm0sYNCkD5AL6y5U"
                />
              </div>
              <div
                className="media-body pt-sm"
              >
                <div
                  className="text-bold"
                >
                  Add class
                  <div
                    className="text-sm text-muted"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-8"
      >
        <div
          className="panel panel-default"
        >
          <div
            className="panel-body"
          >
            <div
              className="pull-right"
            >
              <ul
                className="dropdown-menu dropdown-menu-right animated fadeInLeft"
                role="menu"
              >
                <li>
                  <a
                    href="#"
                  >
                    <span>
                      Send by message
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    <span>
                      Share contact
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    <span>
                      Block contact
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    <span
                      className="text-warning"
                    >
                      Delete contact
                    </span>
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="h4 text-center"
            >
              Profile Setup
            </div>
            <div
              className="row pv-lg"
            >
              <div
                className="col-lg-2"
              />
              <div
                className="col-lg-8"
              >
                <hm
                  className="hm-horizontal ng-pristine ng-valid"
                >
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      h="inputContact1"
                    >
                      Name
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        data_testid="3333"
                        id="inputContact1"
                        placeholder="Patrick Star"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      h="inputContact2"
                    >
                      Email
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        id="inputContact2"
                        placeholder="pstar123@ccny.cuny.edu"
                        type="email"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      h="inputContact3"
                    >
                      Mobile
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        id="inputContact3"
                        placeholder="(123) 465 789"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      htmlFor="inputContact5"
                    >
                      Website
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        id="inputContact5"
                        placeholder="https://bbhosted.cuny.edu/"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      h="inputContact6"
                    >
                      Username
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        data_testid="Patrick"
                        id="inputContact6"
                        placeholder="Star.Patrick"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <label
                      className="col-sm-2 control-label"
                      h="inputContact7"
                    >
                      Password
                    </label>
                    <div
                      className="col-sm-10"
                    >
                      <input
                        className="hm-control"
                        id="inputContact7"
                        placeholder="PatrickIzTheStar"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    className="hm-group"
                  >
                    <div
                      className="col-sm-offset-2 col-sm-10"
                    />
                  </div>
                  <div
                    className="hm-group"
                  >
                    <div
                      className="col-sm-offset-2 col-sm-10"
                    >
                      <button
                        className="btn btn-info"
                        type="submit"
                      >
                        Update
                      </button>
                    </div>
                  </div>
                </hm>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
